### PR TITLE
Print int[] as int array.

### PR DESCRIPTION
**Pull request by nojaf**
Initial attempt to address https://github.com/fsharp/fslang-suggestions/issues/635.
The result of a pair-programming session with @edgarfgp 🎉.

### CODE REVIEW

**Comment by edgarfgp** [comment]
@nojaf As we suspected we have a bunch of test failing now . Good first step. let tackle  them in the next pair session :)

**Comment by edgarfgp** [comment]
We are getting closer :D . We have some baseline tests that I created a PR to you fork to fix them :) 

Some VS integration test are also failing . Im not sure how to fix them . But we can finish that in our next Pair session :)

**Comment by edgarfgp** [comment]
@vzarytovskii  This is ready for a review :) .

**Review by dsyme — APPROVED** [review]

**Comment by dsyme** [comment]
@vzarytovskii I'm happy with this.

**Comment by edgarfgp** [comment]
@vzarytovskii If this could make it to the next VS preview version ? . This will also benefit VSCode and Rider :)

File: src/Compiler/TypedTree/TypedTreeOps.fs [hunk 1]
```diff
@@ -10407,3 +10407,17 @@ let (|EmptyModuleOrNamespaces|_|) (moduleOrNamespaceContents: ModuleOrNamespaceC
         else
             None
     | _ -> None
+
+let (|TTypeMultiDimensionalArrayAsGeneric|_|) (t: TType) =
+    let rec (|Impl|_|) t =
+        match t with
```
Comment (dsyme) [inline src/Compiler/TypedTree/TypedTreeOps.fs:10413]: This code is wrong 

1. once again we're matching directly on TType, we should **almost never** be doing this
2. it looks like it is detecting jagged arrays, which is wrong

File: src/Compiler/Checking/NicePrint.fs [hunk 2]
```diff
@@ -893,6 +903,10 @@ module PrintTypes =
         // Always prefer 'float' to 'float<1>'
         | TType_app (tc, args, _) when tc.IsMeasureableReprTycon && List.forall (isDimensionless g) args ->
           layoutTypeWithInfoAndPrec denv env prec (reduceTyconRefMeasureableOrProvided g tc args)
+        
+        // Special case for nested array<array<'t>> shape
```
Comment (dsyme) [inline src/Compiler/Checking/NicePrint.fs:907]: This is wrong, and this rule should be removed.  Also there are no tests for jagged arrays - they should be added.

File: src/Compiler/TypedTree/TypedTreeOps.fs (hunk 1 above)
Comment (edgarfgp) [reply src/Compiler/TypedTree/TypedTreeOps.fs:10413]: Will have a look next week

File: src/Compiler/Checking/NicePrint.fs (hunk 2 above)
Comment (edgarfgp) [reply src/Compiler/Checking/NicePrint.fs:907]: Will update next week. Thanks for the review

**Pull request merged**